### PR TITLE
Option to toggle hidden folders/files in browser file dialog

### LINKED ISSUE #9215: Filter Hidden files option on browser Open/Save dialog
Hey all,

Recently we have added the option to filter hidden files in the Open/Save browser Dialogs. If you like we can upstream this feature. If not, feel free to close.

Thanks for all of the amazing work you do on Theia!

### Feature Description:

hidden_file_filter

hidden_file_filter_save

### PR DESCRIPTION
**Pull request by kenneth-marut-work**
#### What it does

Resolves #9215 by adding a toggle to show hidden folders and files in the browser's file dialog

image

#### How to test

- Build this branch for browser
- Open a save or open file dialog and test out the toggle
- Does it work as expected?
#### Review checklist

- [x ] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/filesystem/src/browser/file-dialog/file-dialog.ts [hunk 1]
```diff
@@ -174,6 +177,10 @@ export abstract class FileDialog<T> extends AbstractDialog<T> {
         this.locationListRenderer.host.classList.add(NAVIGATION_LOCATION_LIST_PANEL_CLASS);
         navigationPanel.appendChild(this.locationListRenderer.host);
 
+        this.hiddenFilesToggleRenderer = this.hiddenFilesToggleFactory(this.widget.model.tree);
+        console.log('the renderer', this.hiddenFilesToggleRenderer);
```
Comment (colin-grant-work) [inline packages/filesystem/src/browser/file-dialog/file-dialog.ts:181]: Leftover?

**Comment by colin-grant-work** [comment]
Apart from the stray log, this is working well for me. 👍

File: packages/filesystem/src/browser/file-dialog/file-dialog.ts (hunk 1 above)
Comment (kenneth-marut-work) [reply packages/filesystem/src/browser/file-dialog/file-dialog.ts]: yes, thank you. I've removed this

File: packages/filesystem/src/browser/file-dialog/file-dialog-hidden-files-renderer.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    protected init(): void {
+        this.host.classList.add(TOGGLE_HIDDEN_PANEL_CLASS);
+        this.render();
+    }
+
+    protected readonly handleCheckboxChanged = (e: React.ChangeEvent<HTMLInputElement>): void => this.onCheckboxChanged(e);
+    protected override doRender(): React.ReactNode {
+        return (
+            <div className={TOGGLE_HIDDEN_CONTAINER_CLASS}>
+                Show hidden files
```
Comment (vince-fugnitto) [inline packages/filesystem/src/browser/file-dialog/file-dialog-hidden-files-renderer.tsx:43]: Suggestion: 
I believe we should localize the label if possible.

Comment (kenneth-marut-work) [reply packages/filesystem/src/browser/file-dialog/file-dialog-hidden-files-renderer.tsx]: I've added the localization :+1:

**Review by colin-grant-work — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**